src/AppRoutes.jsx

```diff
@@ -52,9 +51,9 @@ const AppRoutes = () => {
       />
       <Route element={<ProtectedRoute isLoggedIn={session.info.isLoggedIn} />}>
         <Route path="/PASS/clients" element={<Clients />} />
-        <Route path="/PASS/documents" element={<Documents />} />
+        <Route path="/PASS/clients/profile" element={<Profile />} />
```

Comment:
I think a better move here would be to use query params. The current system has several client profiles mapping to the same route. That can potentially confuse browser history and other browser caching systems.

We could have the url be something like:
`www.pass.com/profile?webid="some/web/id"` or similar. Then the profile page displays whatever data is fetched from that web ID (we may need to escape the webId).

We could even use this route for the signed in user's profile (github works like this). Then instead of passing a prop to the `Profile` component, we could just check if the webId from the query param is the same as the signed in user's web ID. Something like

```
import React, { useContext } from 'react';
import { useSearchParams } from 'react-router-dom';
import { signedInUserContext} from '@contexts';
import { PersonalProfileComponent, PublicProfileComponent} from '.'

const Profile = () => {
  const { curUserId } = useContext(signedInUserContext);
  const { searchParams } = useSearchParams();
  
   return (
     <> 
       {curUserId === searchParams.get('webId') ? <PersonalProfileComponent/> : <PublicProfileComponent />}
     </>
   )
}
```

This method may make the `SelectedUserContext` unnecessary. If so you can delete it.

Reply:
I'll take a look at dynamic ids for React Router

Reply:
Well, you want to look at url query parameters. That's what this concept is called:
https://www.branch.io/glossary/query-parameters/#:~:text=Query%20parameters%20are%20a%20defined,immediately%20by%20a%20query%20parameter.

Here's how you use query parameters in react router:
https://reactrouter.com/en/6.14.2/hooks/use-search-params

Reply:
Think I got something working using the props being passed into the `ClientListTableRow` (the `labelId`) and using that as the path.